presto-main/src/main/java/com/facebook/presto/operator/scalar/DayOfMonthWithEnglishSuffixPrinter.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import org.joda.time.Chronology;
+import org.joda.time.DateTimeFieldType;
+import org.joda.time.DateTimeZone;
+import org.joda.time.ReadablePartial;
+import org.joda.time.format.DateTimeParser;
+import org.joda.time.format.DateTimeParserBucket;
+import org.joda.time.format.DateTimePrinter;
+import org.joda.time.format.FormatUtils;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Locale;
+
+public final class DayOfMonthWithEnglishSuffixPrinter
+        implements DateTimePrinter, DateTimeParser
+{
+    private static final int MIN_PRINTED_LENGTH = 3;
+    private static final int MAX_PRINTED_LENGTH = 4;
+
+    private final DateTimeFieldType iFieldType;
+
+    public DayOfMonthWithEnglishSuffixPrinter()
+    {
+        iFieldType = DateTimeFieldType.dayOfMonth();
+    }
+
+    @Override
+    public int estimatePrintedLength()
+    {
+        return MAX_PRINTED_LENGTH;
+    }
+
+    private static String getSuffix(int day)
+    {
+        assert day > 0 && day <= 31;
+        switch(day) {
+            case 1:
+            case 21:
+            case 31:
+                return "st";
+            case 2:
+            case 22:
+                return "nd";
+            case 3:
+            case 23:
+                return "rd";
+            default:
+                return "th";
+        }
+    }
+
+    private void printTo(Appendable appendable, long instant, Chronology chrono) throws IOException
+    {
+        try {
+            int day = iFieldType.getField(chrono).get(instant);
+            FormatUtils.appendUnpaddedInteger(appendable, day);
+            appendable.append(getSuffix(day));
+        }
+        catch (RuntimeException e) {
+            appendable.append('\ufffd');
```

Comment:
This is how Joda handles a runtime printing error:
https://github.com/JodaOrg/joda-time/blob/909046aa9aab5fae75e2152fd6ded722a4d5aba2/src/main/java/org/joda/time/format/DateTimeFormatterBuilder.java#L1391